Spiegel Online.js

```diff
@@ -39,64 +39,43 @@ http://www.spiegel.de/international/europe/0,1518,700530,00.html
-	}  else if (doc.location.href.match(/^https?\:\/\/www\.spiegel\.de\/suche/)){ 
-		//Zotero.debug("multiple");
-		return "multiple";
-	}  else if (doc.location.href.match(/^https?\:\/\/www\.spiegel\.de\/international\/search/)){ 
-		//Zotero.debug("multiple");
-		return "multiple";
-	} else if (doc.location.href.match(/^https?\:\/\/www\.spiegel\.de\/international\/topic/)){ 
-		//Zotero.debug("multiple");
+	var spiegel_article_XPath = '//h2[@class="article-title"]';
+	if (doc.location.href.match(/^https?\:\/\/www\.spiegel\.de\/(thema|suche|international\/search|international\/topic)\// )){ 
 		return "multiple";
-	} 
+	} else if (ZU.xpathText(doc, spiegel_article_XPath)) {
+		//the print edition is a magazine. Since the online edition is updated constantly it
+		//makes sense to treat it like a newspaper.
+		if (url.match(/\/print\//)) {
+			return "magazineArticle";
+		} else { 
+			return "newspaperArticle";
+		}
+	}
 }
 
 function scrape(doc, url) {
-	
- 	if (detectWeb(doc, url)=="magazineArticle") {
- 			var newItem = new Zotero.Item("magazineArticle");
- 	}
- 	else{
-		var newItem = new Zotero.Item("newspaperArticle");
- 	}
+	var newItem = new Zotero.Item(detectWeb(doc, url));
+
 	newItem.url = doc.location.href; 
 
 	// This is for the title 
-	
-	var title_xPath = '//div[@class="column-wide"]/h2[contains(@class, "headline")]';
-	if (doc.evaluate(title_xPath, doc, null, XPathResult.ANY_TYPE, null).iterateNext() ){ 
-		var title = doc.evaluate(title_xPath, doc, null, XPathResult.ANY_TYPE, null).iterateNext().textContent;
-		newItem.title = title;
-	} else if (ZU.xpathText(doc, '//div[@id="spArticleColumn"]/h2')) {
-		newItem.title = ZU.xpathText(doc, '//div[@id="spArticleColumn"]/h2');
-	} else {
-		var title = doc.evaluate('//title', doc, null, XPathResult.ANY_TYPE, null).iterateNext().textContent;
-		title = title.split(" - ")[0];
-		newItem.title = title;
-	}
+	newItem.title = ZU.xpathText(doc, '//h2[@class="article-title"]') 
+					|| ZU.xpathText(doc, '//meta[@property="og:title"]/@content') 
+					|| ZU.xpathText(doc, '//title');
+	newItem.title = ZU.trimInternal(newItem.title);
+
 
 	// Tags
-	var tags_xPath = '//meta[contains(@name, "keywords")]';
-	var tags= doc.evaluate(tags_xPath, doc, null, XPathResult.ANY_TYPE, null).iterateNext().content;
-	tags = tags.split(/,/);
-	tags = tags.slice(5); // The first six 5 Tags are generic or section info.
-	if (tags[0] != "" ) {
-		for (var i in tags) {
-			tags[i] = tags[i].replace(/^\s*|\s*$/g, '');
-			newItem.tags.push(tags[i]);
+	var tags = ZU.xpathText(doc, '//meta[contains(@name, "keywords")][1]/@content');
+	//var tags= doc.evaluate(tags_xPath, doc, null, XPathResult.ANY_TYPE, null).iterateNext().content;
+	if (tags && tags.length>0) {
+		tags = tags.split(/,/);
+		tags = tags.slice(5); // The first six 5 Tags are generic or section info.
+		if (tags[0] != "" ) {
+			for (var i in tags) {
+				tags[i] = tags[i].replace(/^\s*|\s*$/g, '');
+				newItem.tags.push(tags[i]);
+			}
```

Comment:
Actually, I don't know why some tags are discarded, because they don't look generic for me, e.g. tags = "EU-Kommission, Greenpeace Energy, Beihilfe, Subventionen, Meiler, AKW, Atomkraft, Wirtschaft, Unternehmen & Märkte, Atomkraftwerke, Anti-Atomkraft-Bewegung, Atomausstieg, Atommüll-Endlager, Ökostrom, Zivile Atomunfälle, Jean-Claude Juncker, Greenpeace"